extensions/pkg/controller/healthcheck/reconciler.go

```diff
@@ -123,6 +123,11 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 				return r.resultWithRequeue(), err
 			}
 		}
+		r.logger.Info("Do not perform HealthCheck for extension resource. Shoot is hibernated.", "name", acc.GetName(), "namespace", acc.GetNamespace(), "kind", acc.GetObjectKind().GroupVersionKind().Kind)
+		return reconcile.Result{}, err
+	}
+	if cluster.Shoot.Status.IsHibernated {
+		r.logger.Info("Do not perform HealthCheck for extension resource. Shoot is waking up.", "name", acc.GetName(), "namespace", acc.GetNamespace(), "kind", acc.GetObjectKind().GroupVersionKind().Kind)
 		return reconcile.Result{}, err
```

Comment:
Isn't this actually the place where we need to use `r.resultWithRequeue()` as the Shoot is waking up and we need to start performing health checks?

Reply:
Or we rely that there will be a update event for the Cluster where the Shoot will have `.spec.hibernation.enabled=false` and `.status.hibernated=false`?

Reply:
Yes, It is watching the Cluster and starts / stops the health checks based on the Shoot hibernation. In case we would miss the one crucial event that sets the status.isHibernated == false, well then it would start whenever there is another change to the Cluster resource - that would typically be during maintenance ? Plus Gardener is able to detect such outdated health check reports. WDYT

Reply:
>  the Shoot is waking up and we need to start performing health checks

I thought it would make sense to start the health checks when the Shoot is fully woken up again (status.isHibernated == false) - otherwise health checks would always fail for the period of time when the reconcile / redeploy of resources has not completed.

Reply:
Is there Cluster update event when the Shoot gets fully woken up?

Reply:
Nice catch @ialidzhikov. Actually, current g/g master doesn't have it, but it will be introduced https://github.com/gardener/gardener/pull/2217. Consequently, the code should be adapted to be backwards compatible.

Reply:
So to be backwards compatible the health checks should start as soon as the Shoot.Spec.Hibernation.Enabled == false?

Reply:
Probably yes, at least for some time. :(

Reply:
I tested it and the health checks do not start after wake up (Cluster Shoot `.status.hibernated=true`)

Reply:
I added a[ remark in the original issue](https://github.com/gardener/gardener/issues/2241#issuecomment-621103812) and will adapt the code.

Reply:
@ialidzhikov ok, due to local setup issues I was only testing on the Cluster resource . It does not start after wakeup due to  #2217 right?

Reply:
Yes

Reply:
Is it ready for review? What was changed to address this comment?

Reply:
Checking the scenarios currently by using it in the aws-provider.

Reply:
Got caught up in something - so the hibernation and wake up scenario look good. However the initial start of the checks does not work when only watching the cluster. I think I have to add a watch on the actual extension resource again.

Reply:
Ok ready for review now. Tested it with provider-aws.